src/providers/leetcode.cr

```diff
@@ -1,5 +1,5 @@
 class LeetcodeStats < SiteStats
-  @@url_regex = Regex.new("https?://(www[.])?leetcode[.]com/(?<username>\\w+)")
+  @@url_regex = Regex.new("https?://(www[.])?leetcode[.]com/u/(?<username>\\w+)")
```

Comment:
I think it wouldn't hurt to keep the old profile URL format since those still work too

```suggestion
  @@url_regex = Regex.new("https?://(www[.])?leetcode[.]com/(u/)?(?<username>\\w+)")
```